internal/tasks/repotester.go

```diff
@@ -53,6 +60,10 @@ func (r *RepoTester) TestRepositories(ctx context.Context) {
 			return
 		}
 
+		eventCallback := func(ctx context.Context, resourceKind api.ResourceKind, orgId uuid.UUID, name string, created bool, updates *api.ResourceUpdatedDetails, err error) {
+			service.EventRepositoryAccessible(ctx, resourceKind, orgId, name, created, updates, err, r.serviceHandler.CreateEvent)
+		}
```

Comment:
Do the same store/service pattern - ReplaceRepositoryStatus should have a callback that gets the old and new status, do a diff, and emit events.